### PR TITLE
#18356 Added description of requires boolean in package-lock.json to …

### PR DESCRIPTION
**Pull request by kreig303**
…npm docs.

Referenced in #18356.

### CODE REVIEW

File: doc/files/package-lock.json.md [hunk 1]
```diff
@@ -46,6 +46,9 @@ The version of the package this is a package-lock for. This must match what's in
 An integer version, starting at `1` with the version number of this document
 whose semantics were used when generating this `package-lock.json`.
 
+### requires
```
Comment (kevinburke) [inline doc/files/package-lock.json.md:49]: I'm not sure that this is accurate, here's what the package-lock.json file looks like for me (generated using 5.4.2):

```patch
     "@types/supertest": {
       "version": "2.0.3",
       "resolved": "https://registry.npmjs.org/@types/supertest/-/supertest-2.0.3.tgz",
       "integrity": "sha512-QxgjuDhQEq4tPJXTOnGoN4P/BWwvftvkaCZhXMp1C1Otk9kGXb7l/wA7kwz3MwtEMnovdYikS+ZtvaQ4RbhF4g==",
-      "dev": true
+      "dev": true,
+      "requires": {
+        "@types/superagent": "3.5.5"
+      }
     },
```

In other words, it's an object, not a boolean.

Comment (kevinburke) [reply doc/files/package-lock.json.md:49]: My apologies, it looks like it's specified in the file in two different places; at the top level, it's a boolean, inside of a package descriptor, it's an object. Not sure the best way to handle that.

Comment (kreig303) [reply doc/files/package-lock.json.md:49]: So how should we handle this?  Some docs are better than none... and if this isn't to be accepted I'll just close it.

**Comment by kreig303** [comment]
Closing.  Out of sync and need to properly PR.

**Pull request closed without merging**